pkg/retry/retry.go

```diff
@@ -58,7 +58,7 @@ func OnError(
 				return nil
 			}
 
-			if retryCount >= retryLimit {
+			if retryLimit > 0 && retryCount >= retryLimit {
```

Comment:
I'm a bit confused. So if `retryLimit == 0`, it'll always keep retrying?

I have a follow-up question if that's the case, but wanted to understand the reasoning & approach here first.

Reply:
Yes retryLimit == 0 will keep retrying.

Reply:
What about using -1 for "infinity" instead of 0, otherwise, there's no way to specify an actual retry limit of 0?